pkg/driver/distro/distro.go

```diff
@@ -218,6 +219,17 @@ func Download(ctx context.Context,
 		printer.Logger.Info("Trying to download a driver.", printer.Logger.Args("url", url))
 
 		req, err := http.NewRequestWithContext(ctx, http.MethodGet, url, nil)
+		if httpHeaders != "" {
```

Comment:
Also, i'd move this check below the `if err != nil` check: no need to set headers to a possibly nil `req`.